### PR TITLE
Alternate settings implementation

### LINKED ISSUE #16: Settings Button

### PR DESCRIPTION
**Pull request by S1robe**
# Description

This change converts SettingsManger to a yml extension. Which in simpler terms is a static file that has helper methods.
This also converts welcome_page.dart to support the new settings implementation and includes the required dialogues according to the design document.

In the comments below is a video procedure for how to update code coverage.
This procedures is as follows:
1. Run integration tests locally, which generates a lcov.info files (code coverage)
2. Commit this file to the repo (which then runs the CI process)
3. CI process runs and merges the Unit tests coverage with the already existing code coverage (which will add in the coverage that integration tests don't cover)
4. Coverage report is uploaded and analyzed by CodeCov, which then delivers the report.

Fixes #16 #14 #76 #67 #58 #49 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

This has been tested extensively, all the previous files that were in “tests/*” were being treated as “Unit tests”, some of which were not, and required a UI, These are considered “integration tests” by flutter.

Integration tests do not run in github actions for various reasons, mainly, the runner does not support haxm or hardware emulation. 

See the files from “integration_tests/*”
- [x] welcome_test.dart - tests the new welcome page
- [x] settings_test.dart - tests the new settings implementation.

**Test Configuration**:
* Firmware version: Google API system_34
* Hardware: N/a
* Toolchain: x86_64
* SDK: API 34

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### Media
image
image
image
image
image

### CODE REVIEW

**Comment by S1robe** [comment]
This will be the new method for uploading code coverage.
Integrations will not run online with github actions. As shown by the 400 commits from this branch
Please follow this guide from now on.

If you have questions please reach out.

https://github.com/S1robe/Pocket-Therapist/assets/78630000/ab158296-db57-402e-a4d2-14257c806a31

**Review by S1robe — COMMENTED** [review]
Reviewed, please let me know how I can improove

**Review by Benek6h — APPROVED** [review]
A long, fought battle.
Thank you for the good work, Garrett!

File: app/lib/provider/encryptor.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+String _keyCipherText= "";
+
+void setPassword(String password) {
+  //Do hash things
+
+  _passwordHash = password;
+}
+
+/// Password can be empty, this is only used if the password is supplied
+bool validatePasswordField(String password) => (password.length >= 10 &&                       // length must be 10+      AND
```
Comment (JikoJeans) [inline app/lib/provider/encryptor.dart:18]: Super interesting work going on with the password validation field, thank you for creating a way to add password parameters ensuring passwords made are secure and for commenting them to make it easy to follow.

**Review by JikoJeans — APPROVED** [review]
Thank you for staying up late to get these changes implemented, I appreciate your hard work!
